jme3-plugins-json/src/main/java/com/jme3/plugins/json/Json.java

```diff
@@ -0,0 +1,87 @@
+ *
+ * * Neither the name of 'jMonkeyEngine' nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jme3.plugins.json;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.management.RuntimeErrorException;
+
+/**
+  * A json parser factory that allows you to set the parser to use.
+  * @author Riccardo Balbo
+  */
+ public class Json {
+    public static final String PROPERTY_JSON_PARSER_IMPLEMENTATION = "com.jme3.JsonParserImplementation";
+    private static final Logger LOGGER = Logger.getLogger(Json.class.getName());
+    private static final String DEFAULT_JSON_PARSER_IMPLEMENTATION = "com.jme3.plugins.gson.GsonParser";
+
+    private static Class<? extends JsonParser> clazz = null;
+
+    /**
+     * Set the parser to use.
+     * @param clazz a class that implements JsonParser
+     */
+    public static void setParser(Class<? extends JsonParser> clazz) {
+        Json.clazz = clazz;
+    }
+
+
+    /**
+     * Create a new JsonParser instance.
+     * @return a new JsonParser instance
+     */
+    @SuppressWarnings("unchecked")
+    public static JsonParser create() {
+        if (Json.clazz == null) {
+            String className = System.getProperty(PROPERTY_JSON_PARSER_IMPLEMENTATION, DEFAULT_JSON_PARSER_IMPLEMENTATION);
+            try {
+                Json.clazz= (Class<JsonParser>) Class.forName(className);
+            } catch (final Throwable e) {
+                LOGGER.log(Level.WARNING, "Unable to access {0}", className);
+                try{
+                    Json.clazz = (Class<JsonParser>) Class.forName(DEFAULT_JSON_PARSER_IMPLEMENTATION);
+                } catch (Throwable e2) {
+                    throw new RuntimeException("Unable to access default json parser", e2);
+                }
+            }
+        }
+               
+
+        try {
+            return clazz.getDeclaredConstructor().newInstance();
+        } catch (Exception e) {
+            throw new RuntimeException("Could not instantiate JsonParser class " + clazz.getName(), e);
```

Comment:
I think it's better to declare a specified type of the RuntimeException here, for example, IllegalStateException or an IllegalArgumentException.

Reply:
I don't mind changing the exception, but i don't think it would be correct to throw IllegalStateException or IllegalArgumentException in this case

Reply:
The right route was to split this exception and never throw a throwable in a catch block, a catch block should handle program recovery mainly as possible, but you stated that this maneuver will not add a benefit to the project, so I don't have much information to decide what's best for this code, RuntimeException is very different from other checked exceptions so I think throwing it randomly after catching checked exceptions is not good.

Reply:
I've changed the implementation a bit, check if you like this more

Reply:
I will check it now, thanks.